lib/src/common/es6_js_proxy.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2015, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+library polymer.lib.src.common.es6_js_proxy;
+
+import 'dart:html';
+import 'dart:js';
+
+import 'package:polymer/src/common/js_proxy.dart';
+import 'package:polymer_interop/polymer_interop.dart';
+export 'package:polymer_interop/polymer_interop.dart'
+    show convertToDart, convertToJs;
+import 'package:reflectable/reflectable.dart';
+
+import 'behavior.dart';
+import 'declarations.dart';
+import 'reflectable.dart';
+
+/// Mixin this class to get js proxy support! If a [JsProxy] is passed to
+/// [convertToJs] then you will get back a [JsObject] which is fully usable from
+/// JS, but proxies all method calls and properties to the dart instance.
+/// Calling [convertToDart] on that [JsObject] will also return the original
+/// dart instance (not a copy).
```

Comment:
Lets add a note about browser compatibility, and possibly mark it experimental for now?